cpp/test/neighbors/ann_ivf_flat.cuh

```diff
@@ -172,7 +180,7 @@ class AnnIVFFlatTest : public ::testing::TestWithParam<AnnIvfFlatInputs<IdxT>> {
         auto half_of_data_view = raft::make_device_matrix_view<const DataT, IdxT>(
           (const DataT*)database.data(), half_of_data, ps.dim);
 
-        auto index_2 = ivf_flat::extend(handle_, index, half_of_data_view);
+        ivf_flat::extend(handle_, &index, half_of_data_view);
```

Comment:
Not likely related to the current test failures, but the original intent of this was to test both variants of the `extent`  api (make new extended index or modify the given index). I think it makes sense to keep this logic, especially given that now these are actually two different codepaths in detail namespace.